app/src/main/java/com/example/nvdovin/weatherapp/factory/GreenDaoFactory.java

```diff
@@ -30,8 +34,19 @@ public void insert(City city) {
         daoSession.getCityDao().insertOrReplace(city);//TODO implement uniqueId logic
     }
 
-    public List<City> loadCities(){
-       return daoSession.getCityDao().loadAll();
+    public List<City> loadSortedCities(SortQueryBuilder<Property>... queryBuilders) {
+
+        if (null != queryBuilders) {
+            QueryBuilder qb = daoSession.getCityDao().queryBuilder();
+            for (SortQueryBuilder sortQueryBuilder : queryBuilders) {
+                if (sortQueryBuilder.isAscending()) {
+                    return qb.orderAsc(sortQueryBuilder.getProperty()).list();
```

Comment:
according to clean code rules, we should have only 1 enter and 1 exit points for a method (currently you have 3 exit points, refactor this please.)
also because you have more exit points, using of a list is useless, as only the first item will be analyzed.